contrib/plugins/qt-creator-project.py

```diff
@@ -0,0 +1,229 @@
+    outFile.write(' <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Build</value>\n')
+    outFile.write(' <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>\n')
+    outFile.write(' <value type="QString" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.BuildSteps.Build</value>\n')
+    outFile.write('</valuemap>\n')
+    outFile.write('<value type="int" key="ProjectExplorer.BuildConfiguration.BuildStepListCount">1</value>\n')
+    outFile.write('<value type="bool" key="ProjectExplorer.BuildConfiguration.ClearSystemEnvironment">false</value>\n')
+    outFile.write('<valuelist type="QVariantList" key="ProjectExplorer.BuildConfiguration.UserEnvironmentChanges">\n')
+    outFile.write(' <value type="QString">' + "PATH=" + os.environ['PATH'] + '  </value>\n')
+    outFile.write('</valuelist>\n')
+    outFile.write('<value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Vorgabe</value>\n')
+    outFile.write('<value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName">' + name + '</value>\n')
+    outFile.write('<value type="QString" key="ProjectExplorer.ProjectConfiguration.Id">GenericProjectManager.GenericBuildConfiguration</value>\n')
+    outFile.write('<valuelist type="QVariantList" key="UserStickyKeys">\n')
+    outFile.write(' <value type="QString">ProjectExplorer.BuildConfiguration.UserEnvironmentChanges</value>\n')
+    outFile.write(' <value type="QString">UserStickyKeys</value>\n')
+    outFile.write('</valuelist>\n')
+    outFile.write('</valuemap>\n')
+
+def addBuildSteps(outFile, buildMeFile):
+    addBuildConfig(outFile, 0, "Bob dev ", "" , buildMeFile)
+    addBuildConfig(outFile, 1, "Bob dev (force)", "-f", buildMeFile)
+    addBuildConfig(outFile, 2, "Bob dev (no checkout)", "-b", buildMeFile)
+    addBuildConfig(outFile, 3, "Bob dev (no deps)", "-n", buildMeFile)
+    addBuildConfig(outFile, 4, "Bob dev (no checkout, no deps)", "-nb", buildMeFile)
+
+    outFile.write('<value type="int" key="ProjectExplorer.Target.BuildConfigurationCount">5</value>\n')
+    outFile.write('<valuelist type="QVariantList" key="UserStickyKeys">\n')
+    outFile.write(' <value type="QString">ProjectExplorer.BuildConfiguration.BuildStepListCount</value>\n')
+    outFile.write(' <value type="QString">ProjectExplorer.BuildConfiguration.UserEnvironmentChanges</value>\n')
+    outFile.write(' <value type="QString">ProjectExplorer.ProjectConfiguration.DisplayName</value>\n')
+    outFile.write('</valuelist>\n')
+
+def generateQtProject(package, destination, updateOnly, args):
+    project = "/".join(package.getStack())
+
+    dirs = []
+    getCheckOutDirs(package, dirs)
+    projectDir = project.replace('/','_')
+    if projectDir[-1:] == '_':
+        projectDir =projectDir[:-1]
+
+
+    if not destination:
+        destination = os.path.join(os.getcwd(), projectDir)
```

Comment:
Maybe use `os.path.join(os.getcwd(), "project", projectDir)` to group them by default in a sub directory?

I would also suggest to use just `package.getName()` for `projectDir`. Right now the name will get quite long if you build something deep down in the hierarchy.